### ISSUE TITLE
Incredibly long time to backfill repos.

### ISSUE THREAD

**Issue by shardern** [atlassian/github-for-jira#1708]
Hi,

I have seen a few other comments with the same issue, however none seem to be on the same scale as our problem.

We migrated to Atlassian Cloud a couple of weeks ago, installed the Jira <> Github plugin and started backfilling things. A couple weeks later and its still backfilling. We are currently on 966/4692 repos.

We have just noticed as well that some repos have automatically generating commits, so wonder if that might be the issue.

Right now, we are wondering the following:
- What happens when we restart the backfill? Does it clear everything it has already done and start from scratch? Would data be lost? Would users notice that links are missing?
- Can you check the progress of a backfill, or are we just flying blind and hoping that it finishes eventually.

Thanks,

Sam Hardern

**Comment by shardern** [comment]
Hi again, 

Another week later and our backfill is still at 966. Any help that can be given on this would be much appreciated.

Thanks,

Sam

**Comment by rachellerathbone** [comment]
Hey @shardern. I've just updated our [FAQs doc](https://support.atlassian.com/jira-cloud-administration/docs/connect-a-github-enterprise-server-account-to-jira-software/) which will hopefully give you some more insight into this issue. Please let me know if you have more questions.

**Comment by shardern** [comment]
Hi,

Thanks for sending that. Good to know that our users can create new links and they will appear. Any idea if there is a way to track specific which repo is being backfilled at the moment? It might help us identify which repo the backfill is stuck on.

Additionally, can you clarify what happens to existing links if we restart the backfill? Does it clear everything it has already filled, or will that data be retained? 

Thanks for the help.

**Comment by ctolley2** [comment]
Hi @rachellerathbone - to add to Sam's question, I saw in the instructions that if the backfill gets stuck, we should identify the large repo it's stuck on, remove it, and then restart the backfill. How can we know which repo the backfill is currently stuck on, are there logs anywhere for this?

What we're struggling with, is that we can restart the backfill but:
1. Is that going to wipe all Github->Jira links and recreate them again, which is an issue for our users
2. Is the backfill going to get stuck again at the same point
3. If yes, how can we identify and remove the repo its stuck on

**Comment by mboudreau** [comment]
We've started doing some preliminary work on making the backfill more visible to our users.  The cause of slowness is 2 pronged:
1 - Github API has a throttle limit so we can't get everything right away and need to wait.  This includes very large and old repositories with thousands if not hundreds of thousands of commits.
2 - For some larger organizations with large repos, the Github API will just up and fail for no reason and return a 500.  These needs to be investigated by github themselves to figure out a fix.  Sadly, our backfill process will get blocked on these as it wants to get all the data our of all repos.

We're working on making everything more transparent and easier to raise a case with Github when something does go wrong on their end.

**Comment by shardern** [comment]
Hi @mboudreau - Thanks for the response. I'd be surprised at this point if this particular repo was so large it was still being backfilled more than a month later (although not outside the realm of possibility). I will contact our Github team and see if they think it would be worth contacting Github in case its the second option that we have encountered.

Looking forward to more features being added to the tool as its a great connector used by lots of our users.

Thanks

**Comment by liskoland** [comment]
Hi, 
we have the same issue with to long backfill of our github repos. even if we backfill only 24 hours back It takes like 5 days to backfill the data and meanwhile we lose the historical data in the development field for all tickets. 
This is a huge issue for our developers and it is a showstopper for some of their work. 
Also, some of the automations  in Jira had to be stopped due to wrong status of PRs shown in Jira, but different (correct) in Github. 
any advise how to fix that and what we can do to make sure our data is current and accurate will be highly appreciated. not to mention that the backfill needs to be automatic - when there is a change - show it in a prompt manner.

**Comment by hkrieg01** [comment]
Hello,
We have the same problem. Is there already a solution foreseeable or is there a workaround?
Kind Regards
Holger

**Comment by BastianSperrhacke-Otto** [comment]
Hello. any progress or workarounds here?

**Comment by hkrieg01** [comment]
Any update on this issue???

**Comment by hkrieg01** [comment]
Is there any chance that this bug will be fixed in a timely manner?

**Comment by bgvozdev** [comment]
Hey folks, we have made significant improvements in backfill performance during the past several months. Could you please try to restart backfill and report back how it went?

<img width="430" alt="image" src="https://github.com/atlassian/github-for-jira/assets/20631664/0249aacd-1516-492b-8700-5d6bdb00307a">

<img width="430" alt="image" src="https://github.com/atlassian/github-for-jira/assets/20631664/23dc7658-e1dc-4065-80a7-c1bdf97d2b5c">

If you still have troubles, please don't hesitate to re-open the issue and provide your Jira hostname so we can troubleshoot on case-by-case basis.
